templates/hooks/eventsourcemapping/sdk_update_post_build_request.go.tpl

```diff
@@ -0,0 +1,9 @@
+
+    // We need to carefully craft the update request if a user
+    // wants to delete their filterCriterias. Mainly because the
+    // aws-sdk-go doesn't try to update nil fields.
```

Comment:
I'm thinking that this is something we should handle for all CRs that have nested structs fields...